doc/man3/SSL_get_session.pod

```diff
@@ -26,7 +26,30 @@ count of the B<SSL_SESSION> is incremented by one.
 =head1 NOTES
 
 The ssl session contains all information required to re-establish the
-connection without a new handshake.
+connection without a full handshake for SSL versions <= TLSv1.2. In TLSv1.3 the
+same is true, but sessions are established after the main handshake has occurred.
+The server will send the session information to the client at a time of its
+choosing which may be some while after the initial connection is established (or
+not at all). Calling these functions on the client side in TLSv1.3 before the
+session has been established will still return an SSL_SESSION object but it
+cannot be used for resuming the session. See L<SSL_SESSION_is_resumable(3)> for
+information on how to determine whether an SSL_SESSION object can be used for
+resumption or not.
+
+Additionally, in TLSv1.3, a server can send multiple session messages for a
+single connection. In that case the above functions will only return information
+on the last session that was received.
```

Comment:
There doesn't seem to me to be a great benefit in deliberately breaking `SSL_get_session()` et al. It still looks like a useful API even if you are using `SSL_CTX_sess_set_new_cb` - i.e. record somewhere that connection is now resumable in the callback, but come back later to actually get the `SSL_SESSION`, without having to up the ref count and store the whole object somewhere else. Or as another example, you just want to get the SESSION back that you set earlier (e.g. to query the max_early_data for the session) etc

I'm not really convinced that breaking it will get everyone using `SSL_CTX_sess_set_new_cb`. If people are calling `SSL_get_session` immediately after `SSL_connect` then its going to fail to resume most of the time anyway. There won't be an obvious error in either case (SSL_get_session deliberately broken or SSL_get_session not broken but returning non-resumable session most of the time).